src/main/java/justin/Parser.java

```diff
@@ -0,0 +1,182 @@
+     */
+    public static Command parse(String message) throws DukeException {
+        Scanner sc = new Scanner(message);
+        String first = sc.next();
+        switch (first) {
+            case "bye": {
+                return new ExitCommand();
+            }
+            case "list": {
+                return new ListCommand();
+            }
+            case "todo": {
+                String description = sc.nextLine();
+                return parseToDo(description);
+            }
+            case "deadline": {
+                String description = sc.nextLine();
+                return parseDeadline(description);
+            }
+            case "event": {
+                String description = sc.nextLine();
+                return parseEvent(description);
+            }
+            case "mark": {
+                int num = sc.nextInt();
+                return new MarkCommand(num);
+            }
+            case "unmark": {
+                int num = sc.nextInt();
+                return new UnmarkCommand(num);
+            }
+            case "delete": {
+                int num = sc.nextInt();
+                return new DeleteCommand(num);
+            }
+            case "find": {
+                String description = sc.nextLine();
+                return parseFind(description);
+            }
+            default: {
+                throw new DukeException("OOPS! Sorry I do not know what that means...");
+            }
+        }
+    }
+
+    /**
+     * Returns a Command through scanning the text file in String format.
+     * @param message The description of the command.
+     * @return A Command.
+     * @throws DukeException if the file has been corrupted.
+     */
+    public static Command parseCommand(String message) throws DukeException {
+        String[] arr = message.split("\\s\\|\\s");
+        if (arr.length == 1) {
+            return new NewCommand();
+        }
+        String task = arr[0];
+        boolean isDone = arr[1] == "X" ? true : false;
+        String description = arr[2];
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MMM d yyyy hhmma");
+        LocalDate ld = null;
+        LocalTime lt = null;
+        switch (task) {
+            case "T":
+                return new AddToDoCommand(description, isDone);
+            case "D":
+                ld = LocalDate.parse(arr[3], formatter);
+                lt = LocalTime.parse(arr[3], formatter);
+                return new AddDeadlineCommand(description, isDone, ld.toString(), lt.toString());
+            case "E":
+                ld = LocalDate.parse(arr[3], formatter);
+                lt = LocalTime.parse(arr[3], formatter);
+                return new AddEventCommand(description, isDone, ld.toString(), lt.toString());
+            default:
+                throw new DukeException("OOPS, looks like the file has been corrupted. Please delete the file and try again.");
```

Comment:
Minor nit, this line seems to be like 6 chars too long, maybe can instantiate the error message separately on another line?